framework/src/media/audio/resample/samplerate.c

```diff
@@ -474,6 +458,9 @@ int src_simple(src_handle_t handle, src_data_t *src_data)
 		RETURN_VAL_IF_FAIL((src->lastnewformat == new_sample_width), SRC_ERR_NOT_SUPPORT);
 	}
 
+	src->out_buffer = src_data->data_out;
+	src->outsize_in_frames = src_data->out_buf_length / (BYTES_PER_SAMPLE(new_sample_width) * channels_num);
```

Comment:
How about checking the validity of "src_data->out_buf_length" in a proper position?

Reply:
out_buffer and out_buf_length are user input values now, 
resampler record them and convert length bytes to frames, 
it's easy for us to use frame conceptual value :), I think.

Reply:
sure, validation checking is necessary, we will fix it.